docs/architecture-decisions/multiple-sync-sources.md

```diff
@@ -0,0 +1,82 @@
+* Decouple storage from flagd entirely, by exposing a Golang interface type that "providers" can implement to provide support for any data store.
+
+## Proposal
+<!--
+Unsure whether we want a diagram in this section or not. Happy to add one if we want one.
+-->
+The solution to the conundrum posited in the background section of this document is to decouple flag storage entirely from the rest of the runtime, including instead support for myriad commonly used data synchronization interfaces. This allows Flagd to be agnostic to flag storage, while enabling users to use whichever storage back-end best suits their environment.
+
+To extend Flagd to support a new storage back-end, _sync providers_ implement the _ISync_ interface, detailed below:
+```
+type ISync interface {
+	// Init is used by the sync provider to initialize its data structures and external dependencies.
+	Init(ctx context.Context) error
+
+	// Sync is the contract between Runtime and sync implementation.
+	// Note that, it is expected to return the first data sync as soon as possible to fill the store.
+	Sync(ctx context.Context, dataSync chan<- DataSync) error
+
+	// ReSync is used to fetch the full flag configuration from the sync
+	// This method should trigger an ALL sync operation then exit
+	ReSync(ctx context.Context, dataSync chan<- DataSync) error
+
+	// IsReady shall return true if the provider is ready to communicate with the Runtime
+	IsReady() bool
+}
+```
+Synchronization events "fan-in" from all configured sync providers to flagd's in-memory state-store via a channel carrying [`sync.DataSync`](https://github.com/open-feature/flagd/blob/main/core/pkg/store/flags.go#L19) events. These events detail the source and type of the change, along with the flag data in question and are merged into the currently held state by the [store](https://github.com/open-feature/flagd/blob/main/core/pkg/store/flags.go#L19).
+
+### API changes
+
+<!--
+This section should describe any API changes that are part of the proposal.
+This includes any new endpoints, changes to existing endpoints, or modifications to the data model.
+It should provide enough detail for developers to understand how the API will evolve and what impact it will have on existing clients.
+-->
+
+Because synchronization providers may vary wildly with respect to their implementation details, supporting multiple synch providers means supporting custom configuration parameters for each provider. As a consequence, Flagd's configuration is itself made more complex, and its bootstrap process, whose goal is to create a [`runtime.Runtime`](https://github.com/open-feature/flagd/blob/main/flagd/pkg/runtime/runtime.go#L21) object from user-provided configuration, spends the preponderance of its time and effort interpreting, configuring, and initializing sync providers. There is, in fact, a custom bootstrap type, called the `syncbuilder` whose job is to bootstrap synch providers and arrange them into a map, for the runtime to use.
```

Comment:
```suggestion
Because synchronization providers may vary wildly with respect to their implementation details, supporting multiple sync providers means supporting custom configuration parameters for each provider. As a consequence, Flagd's configuration is itself made more complex, and its bootstrap process, whose goal is to create a [`runtime.Runtime`](https://github.com/open-feature/flagd/blob/main/flagd/pkg/runtime/runtime.go#L21) object from user-provided configuration, spends the preponderance of its time and effort interpreting, configuring, and initializing sync providers. There is, in fact, a custom bootstrap type, called the `syncbuilder` whose job is to bootstrap sync providers and arrange them into a map, for the runtime to use.
```